### DISCUSSION TITLE
ERR: buffer length exceeds 64

### DISCUSSION THREAD

**Discussion by mister-good-deal** [bluenviron/mediamtx#391] [Questions]
Hello,

I'm trying to broadcast an RTSP stream throw **rtsp-simple-server** in Docker  environement.

I have a container which running an app delivering H264 encoded frame throw NVIDIA **NVENC** encoder.

Each frame is written to a `stream.h264` file and streamed with **FFMPEG** with `AVPacket` and `av_write_frame` to the streaming server container.

The `stream.h264` is readable by VLC after the streaming session.

I initialize the FFMPEG streaming session to an URL `tcp://172.20.0.4:8554/live.stream` (the rtsp-simple-server container's IP).

Resulting in

```
video-extractor-stream | 2021/05/21 10:00:48 I [0/0] rtsp-simple-server v0.16.0
video-extractor-stream | 2021/05/21 10:00:48 I [0/0] [RTSP] TCP listener opened on :8554
video-extractor-stream | 2021/05/21 10:09:09 I [0/0] [RTSP] [conn 172.20.0.2:52654] opened
```

But with the 1st streamed frame the stream shut down the server with

```
video-extractor-stream | 2021/05/21 10:09:19 I [0/0] [RTSP] [conn 172.20.0.2:52654] ERR: buffer length exceeds 64
video-extractor-stream | 2021/05/21 10:09:19 I [0/0] [RTSP] [conn 172.20.0.2:52654] closed
```

I checked the buffer size of the 1st frame which is 326 bytes.

64 bytes is extremely low and I didn't find an option in `rtsp-simple-server.yml` to increase this value.

Attempt to slice the encoded frames with

```c++
h264Config.sliceMode     = 1;
h264Config.sliceModeData = 1500 - 28;
```
in NVENC config does not seem to work as the 2nd frame's size is 179596 bytes.

Am I missing something ?

**Comment by mister-good-deal** [comment]
Can I send H264 encoded frames one by one to **rtsp-simple-server** with `tcp://172.20.0.4:8554/live.stream` URL or is there another logic to apply ?

**Reply by aler9** [reply]
Hello, TCP is not enough to send frames to the server, you have to inizialize an RTSP session and wrap H264 frames inside RTP packets.

This is a sample command line:

```
ffmpeg -i myinput -f rtsp rtsp://172.20.0.4:8554/live.stream
```

Technical explanation on why it's not possible: H264 frames don't contain timing informations (frames DON'T have a timestamp), while H264 frames wrapped inside RTP packets do contain timing informations. Therefore, sending raw H264 frames with TCP is possible only when the stream doesn't contain any B-frame, but contains only I-frames or P-frames.

[a comment by an automated account is not shown]
